NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -1744,7 +1749,362 @@ def get_path(self, iS, n_points=1000):
         dP = path_2d - np.array([self.__X1[0], 0, self.__X1[2]])
         MM = np.matmul(self.__R.T, dP.T)
         path = MM.T + self.__X1
-        return path
+        return path        
+                    
+    def get_effective_index(self, n, direction, nx, ny, nz):
+        """
+        objective function to find the possible values of 'n' into the direction of 'k'
+    
+        Parameters
+        ------------
+        n: float
+            effective index of refraction for the given direction)
+        direction: 3-dim array of floats
+            the propagation direction (must be normalized to 1)
+        nx: float
+            the index of refraction into the x direction
+        ny: float
+            the index of refraction into the y direction
+        nz: float
+            the index of refraciton into the z direction
+
+        
+        """
+        sx = direction[0]
+        sy = direction[1]
+        sz = direction[2]
+        A = (nx**2 - n**2)*(ny**2 - n**2)*(nz**2 - n**2)
+        B = n**2 *( sx**2 *(ny**2 - n**2)*(nz**2 - n**2) + sy**2 *(nx**2 - n**2)*(nz**2 - n**2) + sz**2 *(nx**2 - n**2)*(ny**2 - n**2))
+
+        return A + B
+
+    def get_polarization(self, n, direction, nx, ny, nz, rounding=20):
+        
+        """
+        returns the polarization of a wave for the direction of propagation 
```

Comment:
Please elaborate how the polarization is defined and what format the output is given.